TACC/Core-Portal
# Task/WP-731: Mutation hook: Trash file/folder

## Overview

We are working on updating our current mutations to utilize TypeScript and React Query instead of React Redux. This should improve the speed and functionality of all portal systems going forward.

## Related

* [WP-731](https://tacc-main.atlassian.net/browse/WP-731)

## Changes

I converted `useTrash.js` to `useTrash.ts` and created a mutation that sends a selected file/folder to the Trash by updating its path.

## Testing

1. Navigate to a file/folder that you wish to throw away.
2. Select the file/folder in question.
3. Click on the Trash button.
4. You should get a notification that the file was moved to the Trash.

## UI

![image](https://github.com/user-attachments/assets/ff31fdf9-ff90-49de-b5a0-21e7af8c83cc)

## Notes
